packages/store/src/StoreCore.sol

```diff
@@ -443,24 +444,40 @@ library StoreCore {
 
   /************************************************************************
    *
-   *    EPHEMERAL SET DATA
+   *    OFFCHAIN SET DATA
    *
    ************************************************************************/
 
   /**
-   * Emit the ephemeral event without modifying storage for the full data of the given tableId and key tuple
+   * Emit StoreSetRecord without modifying storage for the given tableId and key tuple
    */
-  function emitEphemeralRecord(
+  function emitSetRecord(
     bytes32 tableId,
-    bytes32[] memory key,
+    bytes32[] memory keyTuple,
     bytes memory data,
     FieldLayout fieldLayout
   ) internal {
+    if (!Tables.getOffchainOnly(tableId)) {
```

Comment:
do we want/need these checks?

Reply:
this might be another good use case for a "resource type" encoded in the resource selector, i.e. table ID

Reply:
it's pretty important to avoid divergence of offchain and onchain state. Yeah a resource type would be a nice alternative